src/main/app/src/pages/BrowsePage.js

```diff
@@ -29,9 +30,19 @@ const BrowsePage = ({ links, activityType, updateActivityType, updateActivity, u
     settextBoxValue("");
   }
 
-  // This is the function that gets triggered on typing any number in the textfield for GAME filters.
-  const handleGameFilterChange = evt => {
-    updateLinksFilters(Object.assign(linkFilters, {numOfPlayers: evt.target.value}));
+  /** This is the function that gets triggered on typing any value in the filter textfield.
+      It sets the filters differently depending on the activity type selected. */
+  const handleFilterChange = evt => {
+    const value = evt.target.value;
+
+    if (value === "")
+      updateLinksFilters({});
+    else
+      updateLinksFilters(Object.assign(linkFilters, 
+        activityType === "games" ? {numOfPlayers: value} :
```

Comment:
David suggested that we use switch statements from now on, wherever we have these ternary operators. He told us not to chnage previous code that got approved, but try to use switch statements from now on!

Reply:
+1 to using switch statement. Also, if three of you are using the activityType a lot, it is better to make it an enum.

Some benefits of enum:
1. Easy to change value. You are replacing articles with readings everywhere. With enum it is hard to mess up.
2. No need to worry about "invalid string" because it is not possible.

Reply:
Fixed it. I changed all the ternary operators to switch statement. And I'm using javascript's enumeration by object which is what is supported.